File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string, int> hash_map;
    
    for (vector<string> c : clothes) {
        hash_map[c[1]]++;
    }
    
    for (auto pair : hash_map) {
        int num = pair.second;
        answer *= (num + 1);
    }
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
    
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String[] str : clothes) {
            String type = str[1];
            map.put(type, map.getOrDefault(type, 0) + 1);
        }
        
        int answer = 1;
        for (int count : map.values()) {
            answer *= (count + 1);
        }

        return answer - 1;
    }
}